js_modules/dagit/packages/core/src/ui/Filters/FilterDropdown.tsx

```diff
@@ -65,51 +88,111 @@ export const FilterDropdown = ({filters, setIsOpen, setPortaledElements}: Filter
         },
       });
     },
-    [setIsOpen, setPortaledElements],
+    [setFocusedItemIndex, setIsOpen, setPortaledElements],
   );
 
+  React.useLayoutEffect(() => {
+    if (isSearchInputFocused) {
+      inputRef.current?.focus();
+    }
+  }, [isSearchInputFocused]);
+
   const allResultsJsx = React.useMemo(() => {
     if (selectedFilter) {
       return selectedFilter
         .getResults(search)
-        .map((result) => (
+        .map((result, resultIndex) => (
           <MenuItem
             key={`filter:${selectedFilter.name}:${result.key}`}
             onClick={() => selectValue(selectedFilter, result.value)}
             text={result.label}
+            index={resultIndex}
+            menuKey={menuKey}
           />
         ));
     }
     const jsxResults: JSX.Element[] = [];
     filters.forEach((filter) => {
       if (filteredFilters.includes(filter)) {
+        const index = jsxResults.length;
         jsxResults.push(
           <MenuItem
             key={`filter:${filter.name}`}
             onClick={() => {
               setSelectedFilter(filter);
+              setFocusedItemIndex(-1);
             }}
             text={
               <Box flex={{direction: 'row', gap: 12}}>
                 <Icon name={filter.icon} />
                 {filter.name}
               </Box>
             }
+            index={index}
+            menuKey={menuKey}
           />,
         );
       }
       results[filter.name]?.forEach((result) => {
+        const index = jsxResults.length;
         jsxResults.push(
           <MenuItem
             key={`filter:${filter.name}:${result.key}`}
             onClick={() => selectValue(filter, result.value)}
             text={result.label}
+            index={index}
+            menuKey={menuKey}
           />,
         );
       });
     });
     return jsxResults;
-  }, [filteredFilters, filters, results, search, selectValue, selectedFilter]);
+  }, [
+    selectedFilter,
+    filters,
+    search,
+    menuKey,
+    selectValue,
+    filteredFilters,
+    results,
+    setFocusedItemIndex,
```

Comment:
With this many dependencies, is the `useMemo` still useful?

Reply:
Mainly for the case where you're tabbing away from the search bar.

Reply:
Specifically when `isSearchInputFocused` changes, since it's not a dep here we won't re-render. I didn't want that to feel sluggish. Also it's possible down the line that some of the filters are async we don't want to call their search function more than once.